api/v1alpha1/provisioning_types.go

```diff
@@ -110,6 +110,9 @@ type ProvisioningSpec struct {
 	// openshift-machine-api namespace. When set to true, this provisioning
 	// configuration would be used for baremetal hosts across all namespaces.
 	WatchAllNamespaces bool `json:"watchAllNamespaces,omitempty"`
+
+	// IronicBootIsoSource
+	IronicBootIsoSource string `json:"ironicBootIsoSource,omitempty"`
```

Comment:
Not sure adding a new field is the way to go. I'm working on the installer changes and I think the current strategy is to send multiple URLs (live-iso, live-initramfs, live-rootfs, live-iso, openstack qcow2). Once these are set in the ProvisioningConfig, the CBO will pass it without change. Then in the ironic-rhcos-downloader, we'll have to unwrap it. If you want we can collaborate on how to get it done correctly.

Reply:
Eventually the idea is to remove the qcow2 completely.

Reply:
@kirankt there may be some confusion here due to the name of the field/variable.
IronicBootIsoSource simply indicates if the boot iso image will be downloaded, cached, and served from the httpd server on the conductor node (local), or if it will be served by the original httpd server (http).
This is related to this ironic upstream change https://opendev.org/openstack/ironic/commit/4874f3c44e7b49714ca4fb158d2551b8f6f5fee4

Reply:
@kirankt This isn't related to the new approach for using the live-iso+IPA, it's for the existing users of the BMH `format: live-iso` which currently caches images (e.g from assisted installer) by default.

I think probably https://github.com/openshift/ironic-image/pull/169  is sufficient, we'll disable caching unconditionally for that use-case, then this PR could be revisited in future if we discover a need to make it configurable.

Reply:
I realize it's WIP but can we avoid saying Ironic? We try to keep the implementation detail hidden. This field also needs more documentation, and should probably be an enum type like ProvisioningNetwork, instead of an open-ended string.